Autopilot/PathManager/Inc/waypointManager.hpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * Waypoint Manager Methods and Helpers
+ * Author: Dhruv Rawat
+ * Created: November 2020
+ * Last Updaetd: November 2020
+ */
+
+#ifndef WAYPOINT_MANAGER_H
+#define WAYPOINT_MANAGER_H
+
+#include <stdlib.h>
+
+#define ORBIT_FOLLOWING 0
+#define LINE_FOLLOWING 1
+
+#define PI 3.14159265359
+#define MAX_PATH_APPROACH_ANGLE PI/2
+
+#define PATH_BUFFER_SIZE 100
+
+//Waypoint Types
+#define DEFAULT_WAYPOINT 0 // Used to navigate flight path
+#define WAYPOINT_UNUSED 1
+#define HOLD_WAYPOINT 2 // Circle
+
+// Placeholder: 
+typedef struct {
+    /*
+        Contains:
+            long 
+            lat
+            alt
+            heading
+    */    
+} _Gps_Data;
+
+// Stores error codes for the waypoint manager
+enum _WaypointStatus {WAYPOINT_SUCCESS = 0, WAYPOINT_NOT_FOUND, WAYPOINT_PARAMETERS_NOT_DEFIND, UNDEFINED_FALIURE};
+
+// Used in the waypointBufferStatus array to signal which elements are free
+enum _WaypointBufferStatus {FREE = 0, FULL};
+
+//Used to specify the modification type when updating the waypointBuffer array
+enum _WaypointBufferUpdateType {ADD_WAYPOINT = 0, UPDATE_WAYPOINT, INSERT_WAYPOINT, DELETE_WAYPOINT};
+
+
+/*
+* Structure stores information about the waypoints along our path to the destination and back. 
+*/
+typedef struct {
+    struct _WaypointData * next;      // Next waypoint
+    struct _WaypointData * previous;  // Previous waypoint
+    long double latitude;             // Latitude of waypoint
+    long double longitude;            // Longitude of waypoint
+    int altitude;                     // Altitude of waypoint
+    float turnRadius;                 // If turn is commanded, this is the radius of the turn
+    char type;
+    char waypointId;                  // Id of the waypoint
+    char arrayIndex;                  // Position of the waypoint within the waypointBuffer (array)
+} _WaypointData;
+
+/*
+* Structure contains the data that will be returned to the Path Manager state manager. 
+* This data will be used by the PID and coordinated turn engine to determine the commands to be sent to the Attitude Manager.
+*/
+typedef struct {
+    uint16_t desiredHeading;            // Desired heading to stay on path
+    int desiredAltitude;                // Desired altitude at next waypoint
+    long double distanceToNextWaypoint; // Distance to the next waypoint (helps with airspeed PID)
+    _WaypointStatus errorCode;          // Contains error codes
+    bool isDataNew;                     // Notifies PID modules if the data in this structure is new
+    uint32_t timeOfData;                // The time that the data in this structure was collected
+} _WaypointManager_Data;
+
+class WaypointManager {
+    public: 
+        /*
+        * Constructor for this class
+        * 
+        * @param _WaypointData * initialWaypoints -> These waypoints will be used to initialize the waypointBuffer array
+        * @param int numWaypoints -> Number of waypoints being initialized in the waypointBuffer array
```

Comment:
you know what, maybe your param's aren't colorful because you're missing the brackets. 

@param[in] for a variable the module intakes
@param[out] for a variable passed by reference that your module stores return data in.

Reply:
Apparently I needed to put /** when starting the comment instead of /*.